paddle/fluid/inference/api/analysis_predictor.h

```diff
@@ -161,4 +181,62 @@ class AnalysisPredictor : public PaddlePredictor {
   bool status_ir_optim_enabled_{false};
 };
 
+class AnalysisPredictor::Quantizer {
+ public:
+  explicit Quantizer(AnalysisPredictor &predictor,  // NOLINT
+                     const std::shared_ptr<QuantizerConfig> &qconfig)
+      : predictor_(predictor), qconfig_(qconfig) {}
+
+  // Execute full quantization procedure.
+  bool Quantize();
+
+#if PADDLE_WITH_TESTING
+  friend class QuantizerTest;
+#endif
+
+ private:
+  // Run single warmup iteration
+  bool RunWarmup() const;
+  // Gather data from variables and calculate scales for them.
+  bool CalculateScales();
+  // Calculate a scale for tensor based on ScaleAlgo rules.
+  void CalculateSingleScale(const std::string &op_name,
+                            const std::string &conn_name,
+                            const std::string &var_name,
+                            const framework::LoDTensor &var_tensor,
+                            bool is_unsigned);
+  void PrepareArgument() const;
+  bool RunQuantizePasses() const;
+  bool SaveModel() const;
```

Comment:
Discussed by @wzzju @Superjomn @bingyanghuang, `SaveModel()` should not be exposed.

Reply:
> Please fix the conflicts, and maybe you can create an independent PR for `cpu_quantize_placement_pass` and its unit-test to speed up the review and merge progress.

OK, working on it.
I will also remove the `SaveModel()`.